web/pages/profile.tsx

```diff
@@ -0,0 +1,716 @@
+              alignItems="center"
+            >
+              <Text fontSize="2xl" fontWeight="semibold">
+                Profile
+              </Text>
+              {editing == false && (
+                <Button
+                  bgColor="#BCBCBC"
+                  width="6%"
+                  borderRadius="16px"
+                  onClick={() => setEditing(true)}
+                >
+                  Edit
+                </Button>
+              )}
+              {editing == true && (
+                <Stack direction="row">
+                  <Button
+                    bgColor="#DAD8D8"
+                    borderRadius="16px"
+                    onClick={() => setEditing(false)}
+                  >
+                    Cancel
+                  </Button>
+
+                  <Button
+                    bgColor="#BCBCBC"
+                    borderRadius="16px"
+                    onClick={() => setEditing(false)}
+                  >
+                    Save
+                  </Button>
+                </Stack>
+              )}
+            </Stack>
+            <Stack direction="column">
+              <Stack direction="row">
+                <Text fontWeight="semibold">General Information</Text>
+                {editing && (
+                  <Text fontSize="larger" color="red" fontWeight="bold">
+                    *
+                  </Text>
+                )}
+              </Stack>
+
+              <Flex border="1px solid black" padding={10}>
+                <Stack
+                  width="100%"
+                  direction="row"
+                  spacing={10}
+                  alignItems="center"
+                >
+                  <Box bgColor="#C9C9C9" borderRadius="100%" boxSize={36}></Box>
+                  {editing && (
+                    <IconButton
+                      icon={<EditIcon />}
+                      position="absolute"
+                      bgColor="white"
+                      left="100px"
+                      top="260px"
+                      border="1px solid black"
+                      borderRadius="100%"
+                      boxSize={10}
+                      aria-label={""}
+                    ></IconButton>
+                  )}
+                  <Stack direction="column" width="85%" spacing={5}>
+                    <Stack direction="row" spacing={5}>
+                      <Stack direction="column" width="50%">
+                        <Text fontWeight="medium">Name</Text>
+                        {editing && (
+                          <Input
+                            placeholder="Firstname Lastname"
+                            disabled={false}
+                          ></Input>
+                        )}
+                        {!editing && (
+                          <Input
+                            placeholder="Firstname Lastname"
+                            disabled={true}
+                          ></Input>
+                        )}
```

Comment:
Combine these into 1 <Input>, dynamically interpret the disabled field: `disabled: {!editing}`. This goes for all the fields